### PR TITLE
Task and asset tests, bug fixes

### PR DESCRIPTION
**Pull request by dmannarino**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): Tests!

## What is the current behavior?

Many task and asset routes broken.
No asset or task tests.
Errors do not conform to JSend standard.
Issue Number: N/A

## What is the new behavior?

- Fixed task and asset routes.
- Asset and task tests.
- At least some errors do conform to JSend standard.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,29 +1,61 @@
...
+    return JSONResponse(
+        status_code=exc.status_code, content={"status": "fail", "data": exc.detail}
+    )
+
+
+@app.exception_handler(ServerError)
+async def server_error_handler(request: Request, exc: ServerError):
+    return JSONResponse(
+        status_code=exc.status_code, content={"status": "error", "message": exc.detail}
+    )
```
Comment (thomas-maschler) [inline app/main.py:49]: What happens if you make handle the HTTP Exception directly?

File: tests/routes/test_assets.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        "creation_options": {
+            "zipped": False,
+            "src_driver": "GeoJSON",
+            "delimiter": ",",
+        },
+    }
+    create_asset_resp = client.post(
+        f"/meta/{dataset}/{version}/assets", json=asset_payload
+    )
+    assert create_asset_resp.json()["status"] == "fail"
```
Comment (thomas-maschler) [inline tests/routes/test_assets.py:35]: When do we use `fail`, when `failed`. What is the logic between the the two?

File: tests/routes/__init__.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        "other": "string",
+        "learn_more": "string",
+    }
+}
+
+
+def create_version(test_client, dataset, version) -> None:
+    # Create dataset and version
+    dataset_resp = test_client.put(f"/meta/{dataset}", json=generic_dataset_metadata)
+    assert dataset_resp.json()["status"] == "success"
```
Comment (thomas-maschler) [inline tests/routes/__init__.py:31]: To stick to DRY I would make this its own function as well

File: tests/routes/test_tasks.py [hunk 4]
```diff
@@ -1,107 +1,196 @@
...
+    # assert patch_resp.json()["status"] == "success"
+    #
+    # # Verify the asset status is now "failed"
+    # get_resp = client.get(f"/meta/{dataset}/{version}/assets/{asset_id}")
+    # assert get_resp.json()["data"]["status"] == "failed"
+    #
+    # # ... but that the version status is still "saved"
+    # get_resp = client.get(f"/meta/{dataset}/{version}")
+    # assert get_resp.json()["data"]["status"] == "saved"
+
```
Comment (thomas-maschler) [inline tests/routes/test_tasks.py:145]: I think, i did not fully implement this. Will take a look and make it a seperate PR.
Also probably best to make this a seperate test

File: tests/routes/test_assets.py (hunk 2 above)
Comment (thomas-maschler) [reply tests/routes/test_assets.py:35]: As discussed, lets deviate from JSEND and make call this `failed`

File: tests/routes/__init__.py (hunk 3 above)
Comment (dmannarino) [reply tests/routes/__init__.py]: Done.

File: tests/routes/test_assets.py (hunk 2 above)
Comment (dmannarino) [reply tests/routes/test_assets.py]: Done.

File: app/main.py (hunk 1 above)
Comment (dmannarino) [reply app/main.py]: Doing so now. Seems to work. Will remove the other handlers + errors in the future as discussed.

File: app/main.py [hunk 5]
```diff
@@ -1,29 +1,72 @@
...
+        status = "error"
+    return JSONResponse(
+        status_code=exc.status_code, content={"status": status, "message": exc.detail}
+    )
+
+
+@app.exception_handler(RequestValidationError)
+async def rve_error_handler(request: Request, exc: RequestValidationError):
+    return JSONResponse(
+        status_code=400, content={"status": "failed", "message": json.loads(exc.json())}
```
Comment (thomas-maschler) [inline app/main.py:66]: I believe response code for this is currently `422`, right?

Comment (dmannarino) [reply app/main.py]: Fixed.

**Pull request merged**